lib/rules/import.js

```diff
@@ -11,9 +11,13 @@ module.exports = {
         return {
             ImportDeclaration(node) {
                 const importPath = node.source.value;
-
+                const importedModules = node.specifiers.length
+                    ? node.specifiers
+                        .filter((childNode) => !!childNode.imported)
+                        .map((childNode) => childNode.imported.name)
+                    : [];
                 imports
-                    .filter((imp) => imp.check(importPath))
+                    .filter((imp) => imp.check(importPath, new Set(importedModules)))
```

Comment:
creating Set from array is O(n) operation, better to use it outside of the filter func